src/databricks/labs/ucx/hive_metastore/grants.py

```diff
@@ -59,6 +59,8 @@ class Grant:
     udf: str | None = None
     any_file: bool = False
     anonymous_function: bool = False
+    success: int = 1
+    failures: str = "[]"
```

Comment:
why do we need to modify this schema? How can we do this without it?

Reply:
At the moment there is no location to record assessment findings for grants/permissions, and for most entities this is how they are recorded.

Two options for doing it without modifying the schema:

 - Use a table alongside, referring to the grant via a business key. (This doesn't make much sense because all the columns are part of the business key, so it effectively duplicates the table.)
 - Perform the findings logic entirely via views, in the style of `code_patterns` and `misc_patterns`.

Of these, the views approach would preferable even though it means relying on integration tests.